[댓글 기능] 미스터코 미션 제출합니다.

eunsukko opened the pull request:
'이제부터 3주차 커밋입니다' [f8e82d0]

이후 커밋부터 확인해주시면 됩니다..!

감사합니다..! (이후 조금 변경될 수도 있습니다.)

green4469 committed:
[게시글 생성/조회] 무민 미션 제출합니다. (#10)

* ADD : 실습미션

* ADD : 제약조건 (HTML중복제거, livereload)

* ADD : 게시글 작성 페이지

* ADD : 게시글 작성

* ADD : 게시글 목록 조회

* ADD : 게시글 조회

* ADD : 게시글 조회 수정

* MOD : 게시글 쓰기 TDD 수정

* ADD : 게시글 수정 페이지 이동

* ADD : 게시글 수정

* ADD : 게시글 삭제

* MOD : Refactor

* rename test

* do not use setter anymore

* add tests for Article class

* remove duplication in tests for ArticleRepository class

* refactor test codes

* use only necessary annotations from lombok

* remove duplication in htmls

cannot put reference to "mypage.css" in common header (css crash)

* remove duplication in html, by common js references

* make setup method for test code of ArticleController

* do not copy when find all articles from repo

green4469 committed:
[게시판 회원 기능] 무민 미션 제출합니다 (#65)

* fetch upstream

* docs: readme 추가(todo list)

* chore: gradle 종속성 추가

- jpa, h2, mysql 추가

* chore: compile -> implementation 변경

* chore: properties에 mysql 종속성 추가

* feat: database를 사용하도록 변경

-  ArticleRepository를 JpaRepository를 상속받는 인터페이스로 변경
-  Article을 @Entity로 변경
-  사용하지 않는 테스트코드 삭제

* refactor: background-image 에러일 경우 기본 이미지 노출되게 수정

* docs: readme 수정

* feat: 비 로그인일 경우 홈에 GUEST - 로그인 - 회원가입이 뜨도록 변경

* Revert "feat: 비 로그인일 경우 홈에 GUEST - 로그인 - 회원가입이 뜨도록 변경"

This reverts commit 664fba831221ff75695e04e1acf2d57e9786466d.

* feat: 비 로그인일 경우 홈에 GUEST - 로그인 - 회원가입이 뜨도록 변경

- controller 변경

* refactor: html 중복 form 제거

* refactor: Article에 toString 추가

* feat: 비 로그인일 경우 홈에 GUEST - 로그인 - 회원가입이 뜨도록 변경

* feat: 회원가입

- 이름, 비밀번호, 이메일 유효성 검사

* feat: 이메일 중복 체크

* feat: User의 email unique 설정

* feat: 로그인/로그아웃 구현 기능

-spring security 구현

* feat: 로그인 관리 security -> session 변경

* feat: add user-list

* feat: add delete account

* feat: confirm password when signup

* chore: add more works in readme

* refactor: refactoring test codes for AccountController class

* feat: add controller exception handlers and 404 page

* chore: update readme

* refactor: seperate domain into two packages

* refactor: do not use @Data, instead list all sub-annotations, correct
order of declarations in User and Article

* refactor: use h2-memory database in test, mysql in production

* refactor: remove magic number

* refactor: remove @Data in UserForm

* refactor: rename UserForm into UserRequestDto

* refactor: rename AccountController into UserController

* refactor: remove unnecessary details in properties for test

* refactor: remove unnecessary annotation

* feat: add NonAuthInterceptor

* refactor: remove session check using interceptor

* refactor: add update method to entity, repository

* refactor: add unique, not null constraints on entity classes

* refactor: remove constructors taking id as an argument in Entity classes

* refactor: remove setters in entity classes

* refactor: use constant rather than hard-coded string for session key of the logged in user

* refactor: separate packages

* refactor: reformat

* refactor: reformat

green4469 committed:
refactor: return boolean expression directly

green4469 committed:
refactor: use HttpStatus constants instead of magic numbers for HTTP status code

green4469 committed:
refactor: separate packages for dtos, exceptions in service layer

green4469 committed:
refactor: rename controller methods in ArticleController

green4469 committed:
refactor: add update method for article

green4469 committed:
refactor: update article-edit.html article update url

green4469 committed:
chore: update properties file

green4469 committed:
chore: add data initialization file for test

green4469 committed:
refactor: simplify article controller test

green4469 committed:
feat: add oneToMay bidirectional relationship

green4469 committed:
feat: add Comment Entity

green4469 committed:
feat: add createdAt column into Comment

green4469 committed:
feat: add CommentRepository

green4469 committed:
feat: create/read comments

green4469 committed:
feat: delete comment

green4469 committed:
debug: use transactional annotations on update services

green4469 committed:
feat: update comment

green4469 committed:
feat: delete comments when delete its parent article

green4469 committed:
feat: add modify comment frontend

eunsukko committed:
Revert "[게시글 생성/조회/수정/삭제] 미스터코 미션 제출합니다. (#44)"

This reverts commit 2dce0cb1e4c01f5bae9081e3b217b13db0bf97b5.

eunsukko committed:
이제부터 3주차 커밋입니다

eunsukko committed:
feat: change propoerties adapt to my mysql accounts

ssosso reviewed (changes requested):
안녕하세요, 미스터코~ 쏘쏘입니다:) (저와의 코드리뷰에서는 '님' 어미를 사용하지 않으셔도 됩니다)
몇 가지 피드백 남깁니다~ 혹시 추가적으로 피드백이 필요하신 부분을 말씀주세요:)

eunsukko committed:
fix: remove unnecessary files

eunsukko committed:
feat: add TODO.md

eunsukko committed:
fix: fix bug (corrupted id in session.user after update profile)

eunsukko committed:
feat: add logger

eunsukko committed:
refactor: handling session

- prehandling through HandlerMethodArgumentResolver
- gathering logic to LoginService

eunsukko committed:
refactor: redirect using RedirectView

eunsukko committed:
refactor: refactor UserController using UserService

eunsukko committed:
refactor: refactor User and Article CotrollerTests

ssosso reviewed (approved):
미스터코~ DM으로도 말씀드렸는데, Pattern과 jquery는 머지 조건에 포함되어 있지 않은 부분이라 다음 미션에 여유가 있으시면 적용해볼 수 있는지 적용하는 것이 좋을지 검토해주세요:)~
이번 미션 고생 많으셨습니다:)
